### ISSUE TITLE
Support for not installing (or installing in a diff directory) ?

### ISSUE THREAD

**Issue by salcode** [JPry/vvv-base#4]
I'm working on a project that loads the core Wordpress files in a sub-directory (`wp`).

My root directory looks like:

- `index.php`
- `/wp`
- `/wp-content`

However, the WordPress core files are being installed in my root directory on provision.  Is there a setting to prevent this?  ( if not, consider this a feature request 😀  )

**Comment by salcode** [comment]
Could we use [wp core is-installed](https://developer.wordpress.org/cli/commands/core/is-installed/) instead?  Maybe here https://github.com/JPry/vvv-base/blob/master/src/Provisioner.php#L253

**Comment by salcode** [comment]
While 0874363 prevents the files from be added on `provision` (after the initial install), it doesn't prevent the files from being added in the first place.

Maybe an additional parameter like `wp_files` that defaults to `true` but when set to `false` is skipped

**Comment by JPry** [comment]
@salcode using `wp core is-installed` is a good idea, except that command is a little ambiguous at first glance. That command actually does this:

> Determines whether WordPress is installed by checking if the standard database tables are installed.

Maybe an option like `wp_root` would be useful, and it could default to `htdocs/`, but be configurable to something else, like `htdocs/wp/`? What do you think?

**Comment by salcode** [comment]
I agree `wp core is-installed` isn't the best solution (though it is working for me as a kluge right now to stop installing WP core in the default location when I already have it in `/wp`).

The project I'm working on uses [WP Starter](https://github.com/wecodemore/wpstarter) so in this case, I don't want to install WP core at all because it gets installed via Composer.

So for my needs being able to say, "Don't install WP files" is what I need.

`wp_files` with a `bool` value would work for me

The `wp_root` idea is pretty cool.  When WP core is installed in a directory, the `index.php` also needs to be modified.  Based on that I suspect it is unusual to install in a different folder without using some kind of system.  (I've only ever done it with __WP Starter__).

Short version:  My guess is a `bool` value will cover most cases.  I'm okay with `wp_root` instead but if we go this route, it would be nice to be able to feed it a `false` value (or something else) to indicate not to add the core files at all.

**Comment by JPry** [comment]
@salcode I've been giving this a bit of thought, since I want to try to make sure that I can solve legitimate problems (like this one) without introducing more problems.

You mentioned above...
>  I don't want to install WP core at all because it gets installed via Composer.

This particular repo is aiming to be a flexible repo that you can use with a wide variety of WordPress projects in VVV. If it were working in an ideal way with the WP Starter workflow you're describing, what would you expect it to do? Particularly, would you expect it to run `composer install` for you? What else would you expect it to do afterwards?

The main reason I'm thinking along those lines is that this repo also does things like running the WordPress installer (if needed), setting up the Nginx config file, installing plugins and themes (if specified), etc. The goal is to automate as much as possible. In the short term, it would be easy to say that using a different workflow requires you to do a bunch of the steps manually, but that defeats the ultimate purpose of this tool. For that reason I wouldn't want it to be a long-term solution.

I'd love any input or thoughts you have.

**Comment by salcode** [comment]
This may be a case where it doesn't make sense to try and handle this edge-case in this repo.

As you mentioned, this repo does many of the same things that a composer setup would do (e.g. installing plugins, themes).

It seems when running WP Starter, setting up the Nginx config file and creating the database are the two tasks that need to happen.  It may make sense for me to setup a different repo to handle this.

**Comment by JPry** [comment]
@salcode [this issue](https://github.com/Varying-Vagrant-Vagrants/VVV/issues/1194) in the main VVV repo got me thinking about having an option to skip all of the WP setup entirely. If I did an option for skipping the setup of WP, and just allowed for creating the DB and the Nginx config, would that work for this particular scenario of yours as well?

**Comment by salcode** [comment]
I like this solution 👍

**Comment by JPry** [comment]
Awesome! I'll be putting together a release in the next few days. However, I think that doesn't matter to VVV, so it should be usable in the meantime

**Comment by nelga** [comment]
Does this have any relation to giving Wordpress it's own directory? Is there any functionality to do this currently?
https://codex.wordpress.org/Giving_WordPress_Its_Own_Directory

**Comment by salcode** [comment]
@nelga Running WordPress in its own directory was the motivation behind this request.  Specifically, I was using the [WP Starter](https://github.com/wecodemore/wpstarter) project in setting it up.

My entry in `vvv-custom.yml` is

```
sf:
    repo: https://github.com/JPry/vvv-base.git
    hosts:
        - sf.test
    nginx_upstream: php71
    custom:
        wp: false
```

I hope this helps.
